src/raftstore/store/store.rs

```diff
@@ -245,19 +245,22 @@ impl<T: Transport, C: PdClient> Store<T, C> {
             return Ok(());
         }
 
-        // Should we need to check other types for stale message?
-        if msg.get_message().get_msg_type() == MessageType::MsgRequestVote {
-            if let Some(peer) = self.region_peers.get(&region_id) {
-                let from_epoch = msg.get_region_epoch();
-                // Is it necessary to check version too?
-                if from_epoch.get_conf_ver() <
-                   peer.storage.rl().region.get_region_epoch().get_conf_ver() {
-                    warn!("RequestVote with a stale epoch {:?}, ignore it", from_epoch);
-                    return Ok(());
-                }
+        // If we receive a message which peer is not in current region and
+        // epoch is stale, we can ignore it.
+        if let Some(peer) = self.region_peers.get(&region_id) {
+            let from_epoch = msg.get_region_epoch();
+            let conf_ver = peer.storage.rl().region.get_region_epoch().get_conf_ver();
+            let not_exist = util::find_peer(&peer.storage.rl().region, from.get_store_id())
+                                .is_none();
+            // Is it necessary to check version too?
+            // Should we need to check RequestVote even peer in this region too?
+            if from_epoch.get_conf_ver() < conf_ver && not_exist {
+                warn!("RequestVote with a stale epoch {:?}, ignore it", from_epoch);
```

Comment:
btw, in raft, if the peer is not found and the message is response, an error returns.
```
// Step advances the state machine using the given message.
    pub fn step(&mut self, m: Message) -> Result<()> {
        // ignore unexpected local messages receiving over network
        if is_local_msg(&m) {
            return Err(Error::StepLocalMsg);
        }
        if self.raft.prs.contains_key(&m.get_from()) || !is_response_msg(&m) {
            return self.raft.step(m);
        }
        Err(Error::StepPeerNotFound)
    }
```